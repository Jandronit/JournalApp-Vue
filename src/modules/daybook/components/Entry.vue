<template>
<div class="entry-container mb-3 pointer p-2"
     @click="$router.push({ name: 'entry', params: { id: 10 }})">
>
<!--  Tittle-->
  <div class="entry-title d-flex">
    <span class="text-success fs-5 fw-bold">4</span>
    <span class="mx-1 fs-5">Diciembre</span>
    <span class="mx-2 fw-light">2022, Domingo</span>
  </div>
<!--  Content-->
  <div class="entry-description">
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Accusantium, adipisci, alias, amet animi asperiores atque autem beatae consectetur consequatur corporis cumque debitis dignissimos doloremque dolorum ea eius enim eos esse est et eum ex expedita explicabo facilis fuga fugiat fugit harum hic id illum impedit in incidunt inventore ipsa ipsam ipsum iste iure iusto laboriosam laborum laudantium libero magnam magni maiores maxime minima
  </div>
</div>
</template>

<script>
export default {
  name: 'Entry',
}
</script>

<style lang="scss" scoped>

.entry-container{
  border-bottom: 1px solid #2c3e50;
  transition: 0.3s all ease-in;

  &:hover{
    background-color: lighten($color: #2c3e50, $amount: 45%);
    transition: 0.3s all ease-in;
  }
}

.entry-description{
  font-size: 0.75rem;
}
</style>
