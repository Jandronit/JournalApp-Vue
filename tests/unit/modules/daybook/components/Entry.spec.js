import {shallowMount} from '@vue/test-utils'
import Entry from '@/modules/daybook/components/Entry.vue'
import {journalState} from '../../../../mocks/test-journal-state';

describe('Test in component Entry', () => {
    //mockRouter
    const mockRouter = {
        push: jest.fn()
    }
    const wrapper = shallowMount(Entry, {
        props: {
            entry: journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    })

    test('should match with the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('should redirect when clicking on the entry-container', () => {
        wrapper.find('.entry-container').trigger('click')
        expect(mockRouter.push).toHaveBeenCalled()
        expect(mockRouter.push).toHaveBeenCalledWith({
            name: 'entry',
            params: {
                id: journalState.entries[0].id
            }
        })
    })
    test('show the correct computed properties', () => {
        expect(wrapper.vm.day).toBe(13)
        expect(wrapper.vm.month).toBe('Marzo')
        expect(wrapper.vm.yearDay).toBe('2023, Lunes')
    })
})
