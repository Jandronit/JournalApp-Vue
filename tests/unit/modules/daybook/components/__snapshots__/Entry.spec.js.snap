// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component Entry should match with the snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!--  Tittle-->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">13</span><span class="mx-1 fs-5">Marzo</span><span class="mx-2 fw-light">2023, Lunes</span></div><!--  Content-->
  <div class="entry-description">Hola mundo 1</div>
</div>
`;
